src/main/java/spoon/reflect/code/CtRecordPattern.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-License-Identifier: (MIT OR CECILL-C)
+ *
+ * Copyright (C) 2006-2023 INRIA and contributors
+ *
+ * Spoon is available either under the terms of the MIT License (see LICENSE-MIT.txt) or the Cecill-C License (see LICENSE-CECILL-C.txt). You as the user are entitled to choose the terms under which to adopt Spoon.
+ */
+package spoon.reflect.code;
+
+import spoon.reflect.annotations.PropertyGetter;
+import spoon.reflect.annotations.PropertySetter;
+import spoon.reflect.path.CtRole;
+import spoon.reflect.reference.CtTypeReference;
+import spoon.support.UnsettableProperty;
+
+import java.util.List;
+
+// TODO (440) metamodel
+
+/**
+ * This code element defines a record pattern, introduced in Java 21
+ * by <a href=https://openjdk.java.net/jeps/440>JEP 440</a>.
+ * <p>
+ * Example:
+ * <pre>
+ *     Object obj = null;
+ *     boolean longerThanTwo = false;
+ *     record MyRecord(String value) {}
+ *     // MyRecord(var string) is the record pattern
+ *     if (obj instanceof MyRecord(var string)) {
+ *         longerThanTwo = string.length() > 2;
+ *     }
+ * </pre>
+ */
+public interface CtRecordPattern extends CtPattern, CtExpression<Void> {
```

Comment:
Why does this extend `CtExpression<Void>`?

Reply:
This decision is basically copied from `CtTypePattern`. Iirc, we decided to go this route because we needed a `CtExpression` for the `CtBinaryOperator`, but the pattern doesn't really have a type.